src/fastify-plugins/maps/static-maps.js

```diff
@@ -0,0 +1,284 @@
+          fastify.mapeoStaticMaps.getResolvedStyleJson(styleId, serverAddress),
+        ])
+
+        stats = results[0]
+        styleJson = results[1]
+      } catch (err) {
+        throw new NotFoundError(`id = ${styleId}, style.json`)
+      }
+
+      rep.headers({
+        'Content-Type': 'application/json; charset=utf-8',
+        'Cache-Control': 'max-age=' + 5 * 60, // 5 minutes
+        'Access-Control-Allow-Headers':
+          'Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since',
+        'Access-Control-Allow-Origin': '*',
+        'Last-Modified': new Date(stats.mtime).toUTCString(),
+        'Content-Length': Buffer.from(styleJson).length,
+      })
+
+      return styleJson
+    }
+  )
+
+  // Get a tile (extension specified)
+  fastify.get(
+    `/:styleId/tiles/:tileId/:z/:x/:y.:ext`,
+    { schema: { params: GetStaticMapTileParamsSchema } },
+    handleStyleTileGet
+  )
+  // Get a tile (extension not specified)
+  fastify.get(
+    `/:styleId/tiles/:tileId/:z/:x/:y`,
+    { schema: { params: GetStaticMapTileParamsSchema } },
+    handleStyleTileGet
+  )
+}
+
+/**
+ * @param {string} archive
+ * @param {string} filename
+ */
+function extractAsarFile(archive, filename) {
+  try {
+    return asar.extractFile(archive, filename)
+  } catch (err) {
+    return undefined
+  }
+}
+
+const mime = new Mime(standardTypes, { 'application/x-protobuf': ['pbf'] })
+
+/**
+ * @param {string} baseDirectory
+ * @param {import('@sinclair/typebox').Static<typeof GetStaticMapTileParamsSchema>} params
+ * @returns {null | { data: Buffer, mimeType: string | null, shouldGzip: boolean}}
+ */
+function getStyleTileInfo(baseDirectory, params) {
+  const { styleId, tileId, z, x, y } = params
+  let { ext } = params
+
+  // TODO: If necessary, need to flip the y value first if the TileJSON source is TMS scheme
+  // Doing this will depend on if we decide that the asar directory structure should only follow XYZ or if it should align with corresponding tilejson spec
+
+  const fileBasename = path.join(z.toString(), x.toString(), y.toString())
+  const asarPath = path.join(baseDirectory, styleId, 'tiles', tileId + '.asar')
+
+  /** @type {Buffer | undefined} */
+  let data
+
+  if (ext) {
+    data = extractAsarFile(asarPath, fileBasename + '.' + ext)
+  } else {
+    // Try common extensions
+    const extensions = ['png', 'jpg', 'jpeg']
+
+    for (const e of extensions) {
+      data = extractAsarFile(asarPath, fileBasename + '.' + e)
+
+      // Match found, use the corresponding extension moving forward
+      if (data) {
+        ext = e
+        break
+      }
+    }
+  }
+
+  // extension check isn't fully necessary since the buffer will only exist if the extension exists
+  // but useful to check for types reasons
+  if (!data || !ext) {
+    return null
+  }
+
+  const mimeType = mime.getType(ext)
+
+  // Set gzip encoding on {mvt,pbf} tiles.
+  const shouldGzip = /mvt|pbf$/.test(ext)
```

Comment:
Nitpick: This [will match][0] any string that ends with `pbf` or contains `mvt`. I may be wrong, but I don't think that's what you want.

```suggestion
  const shouldGzip = ext === 'mvt' || ext === 'pbf'
```

[0]: https://regex101.com/r/srscys/2

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/435/commits/01e881761320c06d3da6af1346670804fb7145a5